src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -352,11 +380,22 @@ public void validateUserCanRead(
             .orElseGet(() -> new BucketSourceOption[0]);
 
     for (String bucket : buckets) {
-      List<Boolean> permissions =
-          storageAsPet.testIamPermissions(
-              bucket, List.of(GCS_SOURCE_BUCKET_REQUIRED_PERMISSION), options);
+      List<Boolean> permissions;
+      try {
+        permissions =
+            storageAsPet.testIamPermissions(
+                bucket, List.of(GCS_SOURCE_BUCKET_REQUIRED_PERMISSION), options);
+      } catch (StorageException e) {
+        // This is a potential failure mode for permissions checking: not being able to make the
+        // permissions check call at all
+        if (e.getCode() == HttpStatus.SC_FORBIDDEN) {
+          permissions = null;
```

Comment:
Is this explicit null set needed?  Any reason that you didn't write this block as:

```
if (e.getCode() != HttpStatus.SC_FORBIDDEN) {
  throw e;
}
```

Reply:
If you don't set `null` here, you get an error since `permissions` is no longer initialized before use below. That could be avoided by initializing `permissions` when it's declared, but then you're writing to it twice.

In any case I would recommend using `List.of()` instead of `null` to avoid the null check below.
```suggestion
          permissions = List.of();
```

Another approach is to do the check of `permissions.equals(List.of(true))` after line 387 above and return early if true. Then you no longer need `permissions` outside the scope of the `try` and the `if` can be removed.

Reply:
Ah derp, that's a great point